tests/test_transform.py

```diff
@@ -252,6 +255,65 @@ def test_null_object_transform(self):
         empty_data = {'addrs': {}}
         self.assertDictEqual(empty_data, transform(empty_data, schema))
 
+    def test_decimal_types_transform(self):
+        schema =  {"type": "object",
+                   "properties": {"percentage": {"type": ["null", "string"],
+                                                 "format": "singer.decimal"}}}
+
+        inf = {'percentage': 'Infinity'}
+        negative_inf = {'percentage': '-Infinity'}
+        root2  = {'percentage': math.sqrt(2)}
+        nan = {'percentage': decimal.Decimal('NaN')}
+        snan = {'percentage': decimal.Decimal('sNaN')}
+        null = {'percentage': None}
+        self.assertEquals(inf, transform(inf, schema))
+        self.assertEquals(negative_inf, transform(negative_inf, schema))
+        self.assertEquals({'percentage': str(math.sqrt(2))}, transform(root2, schema))
```

Comment:
This should be a hardcoded value instead of `str(match.sqrt(2))` so it's 1) more readable, and 2) less like it's reimplementing the code 😅 .